observation_portal/accounts/serializers.py

```diff
@@ -1,28 +1,49 @@
 from rest_framework import serializers
 from django.contrib.auth.models import User
+from django.utils import timezone
+from django.utils.translation import ugettext as _
 
 from observation_portal.accounts.models import Profile
 
 
 class ProfileSerializer(serializers.ModelSerializer):
+    accept_terms = serializers.BooleanField(write_only=True, required=False)
+    terms_accepted = serializers.DateTimeField(read_only=True)
+
     class Meta:
         model = Profile
         fields = (
             'education_user', 'notifications_enabled', 'notifications_on_authored_only',
-            'simple_interface', 'view_authored_requests_only', 'staff_view', 'title',
-            'institution', 'api_quota'
+            'simple_interface', 'view_authored_requests_only', 'title', 'staff_view',
+            'institution', 'api_quota', 'terms_accepted', 'accept_terms'
         )
 
+    def validate_staff_view(self, staff_view):
+        user = self.context.get('request').user
+        is_staff = user and user.is_authenticated and user.is_staff
+        if staff_view and not is_staff:
+            raise serializers.ValidationError(_('Must be staff to set staff_view'))
+        return staff_view
+
+    def validate_accept_terms(self, accept_terms):
+        if not accept_terms:
```

Comment:
I guess this feels a bit strange that they have to submit `"accept_terms": True` when you can't submit it as False. Maybe it would be better to just have a /api/profile/accept_terms endpoint that sets the field to True, and then the field has no other mechanism of being updated? Maybe that is too convoluted though and this is fine as is....

Reply:
I made the change... As far as I could tell, `APIView`s don't allow me to add another endpoint on with the `action` decorator like we do in our ViewSets, so I added another `APIView` solely for accepting the terms and explicitly mapped it to `/api/profile/accept_terms`.